### PR TITLE
Refactor operation.rs

### LINKED ISSUE #45: Refactor operation.rs
We should:
-  introduce a constructor for `Operation`.
- move out `real_to_complex` to math.rs

### PR DESCRIPTION
**Pull request by BAMF0**
Resolves #45

### CODE REVIEW

File: src/operation.rs [hunk 1]
```diff
@@ -2,18 +2,54 @@ use ndarray::{array, Array2};
...
     targets: Vec<usize>,
-    arity: usize,
+}
+
+impl Operation {
+    /// Constructs an operation with the given matrix, targets and arity.
+    ///
+    /// Returns an operation if:
+    /// - `matrix` is square with sides equal to number of `targets`
+    /// - number of `targets` is equal to `arity`
```
Comment (simonengstrm) [inline src/operation.rs:29]: Arity doesn't need to exist as a parameter here anymore

**Review by simonengstrm — CHANGES_REQUESTED** [review]
Just one place where arity is still used as parameter, no need to exist anymore.

File: src/operation.rs (hunk 1 above)
Comment (BAMF0) [reply src/operation.rs:29]: you do not need to exist anymore!!!!!!!

Comment (BAMF0) [reply src/operation.rs:29]: lmao xd

**Review by simonengstrm — APPROVED** [review]
lgtm

**Pull request merged**